File: bongf/java/src/level2/NLcd.java
package level2;

import java.math.BigInteger;
import java.util.Arrays;

public class NLcd {
    public static void main(String[] args) {

    }

    public int solution(int[] arr) {
        return Arrays.stream(arr)
                .reduce(1, (x, y) -> x * y / gcd(x, y));
    }

    private int gcd(int x, int y) {
        return x % y == 0 ? y : gcd(y, x % y);
    }

    public long solution2(int[] arr) {
       return Arrays.stream(arr)
                .mapToLong(x -> Long.valueOf(x))
                .reduce(1, (x, y) -> x * y / BigInteger.valueOf(x).gcd(BigInteger.valueOf(y)).longValue());
    }
}

File: programmers/python/level2/n_lcd.py
from functools import reduce
def solution(arr):
    gcd = lambda x,y : y if x%y == 0 else gcd(y, x%y)
    lcm = lambda x,y : x*y // gcd(x,y)
    return reduce(lambda x,y: lcm(x,y), arr, 1)

from math import gcd
def solution2(arr) :
    return reduce(lambda x,y: x*y//gcd(x,y), arr, 1)

print(solution2([2,6,8,14]))
print(solution2([1,2,3]))

